File: CS/CS.NET-VB.NET/FormToForm/VB2008FormToFormProperty/VB2008FormToFormProperty/Form1.vb
Public Class Form1

    Friend Property TextBoxText() As String
        Get
            Return textBox1.Text
        End Get
        Set(ByVal Value As String)
            textBox1.Text = value
        End Set
    End Property


    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click

        'Dim F2 As Form2 = New Form2() 
        Dim F2 As Form2 = New Form2(Me) '

        F2.ShowInTaskbar = False
        F2.ShowDialog()

    End Sub

End Class

File: CS/CS.NET/Windows Forms .NET/FormToForm/Property/1/CS2008FormToFormProperty/CS2008FormToFormProperty/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace CS2008FormToFormProperty
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        internal string TextBoxText
        {
            get
            {
                return textBox1.Text;
            }
            set
            {
                textBox1.Text = value;
            }
        }    

        private void button1_Click(object sender, EventArgs e)
        {
            Form2 F2 = new Form2();

            F2.ShowInTaskbar = false;
            F2.ShowDialog();
        }

        //Bug
        private void button2_Click(object sender, EventArgs e)
        {
            //this.ShowInTaskbar = !this.ShowInTaskbar;
            MessageBox.Show("Number of open forms: " + Application.OpenForms.Count);       
        }
    }
}
